packages/vaex-ml/vaex/ml/transformations.py

```diff
@@ -776,6 +777,115 @@ def transform(self, df):
 
         return copy
 
+class KBinsDiscretizer(Transformer):
+    '''Bin continous features into discrete bins.
+
+    A stretegy to encode continuous features into discrete bins. The transformed
+    columns contain the bin label each sample falls into. In a way this
+    transformer Label/Ordinal encodes continous features.
+
+    Example:
+
+    >>> import vaex
+    >>> import vaex.ml
+    >>> df = vaex.from_arrays(x=[0, 2.5, 5, 7.5, 10, 12.5, 15])
+    >>> bin_trans = vaex.ml.KBinsDiscretizer(features=['x'], n_bins=3, strategy='uniform')
+    >>> bin_trans.fit_transform(df)
+      #     x    binned_x
+      0   0             0
+      1   2.5           0
+      2   5             1
+      3   7.5           1
+      4  10             2
+      5  12.5           2
+      6  15             2
+    '''
+    n_bins = traitlets.Int(allow_none=False, default_value=5, help='Number of bins. Must be greater than 1.')
+    strategy = traitlets.Enum(values=['uniform', 'quantile', 'kmeans'], default_value='uniform', help='Strategy used to define the widths of the bins.')
+    prefix = traitlets.Unicode(default_value='binned_', help=help_prefix)
+    n_bins_ = traitlets.Dict(help='Number of bins per feature.').tag(output=True)
+    bin_edges_ = traitlets.Dict(help='The bin edges for each binned feature').tag(output=True)
+
+    def fit(self, df):
+        '''
+        Fit KBinsDiscretizer to the DataFrame.
+
+        :param df: A vaex DataFrame.
+        '''
+
+        # We need at least two bins to do the transformations
+        assert self.n_bins > 1, ' Kwarg `n_bins` must be greated than 1.'
+
+        # Find the extent of the features
+        minmax = df.minmax(expression=self.features)
+        epsilon = 1e-8
+        # minmax[:, 0] = minmax[:, 0] - epsilon
+        minmax[:, 1] = minmax[:, 1] + epsilon
+
+        # Bin enges and number of bins
+        bin_edges = {}
+        n_bins = {}
+
+        # Determine the bin edges:
+        if self.strategy == 'uniform':
+            bin_edges = {feat: np.linspace(minmax[i, 0], minmax[i, 1], self.n_bins+1) for i, feat in enumerate(self.features)}
+
+        if self.strategy == 'quantile':
```

Comment:
```suggestion
        elif self.strategy == 'quantile':
```

Reply:
`strategy` is a `traitlet.Enum` so it can only have the pre-defined 3 values. Given that, I find 3 separate `if` statements easier to read (personally). If you feel strongly about it, I can convert it to `elseif` form

Reply:
Yes, very strongly, this is very uncommon to see. Multiple if statements read as if multiple cases can happen, if/elif/elif/else is basically an exclusive set.

Reply:
Done!